### PR TITLE
adding external oqsprovider testing

### PR DESCRIPTION
**Pull request by baentsch**
As per https://github.com/openssl/openssl/pull/17733#issuecomment-1055389417 adding [oqsprovider](https://github.com/open-quantum-safe/oqs-provider) as a submodule for (external) testing.

### CODE REVIEW

**Comment by romen** [comment]
This already looks very good! Thanks for the PR!

**Comment by romen** [comment]
There are some docs fixes to please the markdown linter, and we should add the oqs tests to the external tests CI, either asp part of the existing external_tests task or as it's own, like we do for pyca.

**Comment by baentsch** [comment]
> There are some docs fixes to please the markdown linter, and we should add the oqs tests to the external tests CI, either as part of the existing external_tests task or as its own, like we do for pyca.

Both done.

File: test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+echo "   SRCTOP:             $SRCTOP"
+echo "   BLDTOP:             $BLDTOP"
+echo "   OPENSSL_ROOT_DIR:   $OPENSSL_ROOT_DIR"
+echo "   OpenSSL version:    $OPENSSL_VERSION"
+echo "------------------------------------------------------------------"
+
+if [ ! -d $SRCTOP/oqs-provider/oqs ]; then
+# disable rainbow family by default; all further config options listed at
+# https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs
+   cd $SRCTOP/oqs-provider && git clone --depth 1 --branch main https://github.com/open-quantum-safe/liboqs.git && cd liboqs && mkdir build && cd build && cmake -DOQS_ENABLE_SIG_RAINBOW=OFF -DCMAKE_INSTALL_PREFIX=$SRCTOP/oqs-provider/oqs .. && make && make install
```
Comment (levitte) [inline test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh:51]: You'll suffer from the effects of unrestricted `cd`.  Suggested alternative (making it a bit more readable while I'm at it):

```suggestion
   (
       cd $SRCTOP/oqs-provider \
           && git clone --depth 1 --branch main https://github.com/open-quantum-safe/liboqs.git \
           && cd liboqs \
           && mkdir build \
           && cd build \
           && cmake -DOQS_ENABLE_SIG_RAINBOW=OFF -DCMAKE_INSTALL_PREFIX=$SRCTOP/oqs-provider/oqs .. \
           && make \
           && make install
   )
```

File: test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+fi
+
+if [ ! -f $SRCTOP/oqs-provider/test/ssltestlib.c ]; then
+   cd $SRCTOP/oqs-provider && OPENSSL_BLDTOP=$BLDTOP ./scripts/preptests.sh
```
Comment (levitte) [inline test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh:55]: You'll suffer from the effects of unrestricted `cd`. Suggested alternative (making it a bit more readable while I'm at it):

```suggestion
   ( cd $SRCTOP/oqs-provider && OPENSSL_BLDTOP=$BLDTOP ./scripts/preptests.sh )
```

**Review by levitte — CHANGES_REQUESTED** [review]

File: test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh (hunk 2 above)
Comment (baentsch) [reply test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh]: Correct observation, thanks. Fixed in https://github.com/openssl/openssl/pull/17832/commits/fed1c5e0d3f006598c1e69f98dc52ba090b0a02a.

File: test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh (hunk 1 above)
Comment (baentsch) [reply test/recipes/95-test_external_oqsprovider_data/oqsprovider.sh]: Correct observation, thanks. Fixed in https://github.com/openssl/openssl/pull/17832/commits/fed1c5e0d3f006598c1e69f98dc52ba090b0a02a.

**Review by levitte — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Squashed the commits and slightly reworded the commit message and merged to master branch. Thank you for your contribution!

**Pull request closed without merging**